File: web/src/api/apiService.ts
import axios from 'axios';

const API_BASE_URL =
  process.env.EXPO_PUBLIC_API_BASE_URL ?? 'http://localhost:3000';

const api = axios.create({
  baseURL: API_BASE_URL,
  timeout: 30000,
  headers: { 'Content-Type': 'application/json' },
});

// Attach auth token to every request
export function setAuthToken(token: string | null) {
  if (token) {
    api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
  } else {
    delete api.defaults.headers.common['Authorization'];
  }
}

// --- Auth ---
export async function login(email: string, password: string) {
  const res = await api.post('/api/auth/login', { email, password });
  return res.data;
}

export async function register(
  name: string,
  email: string,
  password: string,
  age: number,
  gender: string,
  language: string
) {
  const res = await api.post('/api/auth/register', { name, email, password, age, gender, language });
  return res.data;
}

export async function verifyOtp(email: string, otp: string, type: string) {
  const res = await api.post('/api/auth/verify-otp', { email, otp, type });
  return res.data;
}

export async function resendOtp(email: string, type: string) {
  const res = await api.post('/api/auth/resend-otp', { email, type });
  return res.data;
}

export async function forgotPassword(email: string) {
  const res = await api.post('/api/auth/forgot-password', { email });
  return res.data;
}

export async function resetPassword(email: string, otp: string, newPassword: string) {
  const res = await api.post('/api/auth/reset-password', { email, otp, newPassword });
  return res.data;
}

// --- Profile ---
export async function updateProfile(data: {
  name?: string;
  email?: string;
  language?: string;
  district?: string;
  state?: string;
}) {
  const res = await api.put('/api/users/profile', data);
  return res.data;
}

export async function deleteAccount() {
  const res = await api.delete('/api/users/profile');
  return res.data;
}

// --- Content ---
export async function fetchDiseases() {
  const res = await api.get('/api/content/diseases');
  return res.data;
}

export async function fetchLearningContent() {
  const res = await api.get('/api/content/learning');
  return res.data;
}

export async function submitFeedback(name: string, email: string, message: string) {
  const res = await api.post('/api/content/feedback', { name, email, message });
  return res.data;
}

// --- Gemini Chatbot ---
const GEMINI_API_KEY = process.env.EXPO_PUBLIC_GEMINI_API_KEY ?? '';
const GEMINI_URL =
  'https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent';

export interface ChatTurn {
  role: 'user' | 'model';
  parts: Array<{ text: string }>;
}

export async function sendGeminiMessage(
  history: ChatTurn[],
  newMessage: string
): Promise<string> {
  const key = GEMINI_API_KEY;
  if (!key) {
    return 'Chatbot is not configured. Please add EXPO_PUBLIC_GEMINI_API_KEY to web/.env and restart.';
  }

  const systemPrompt =
    "You are ORCare AI, a friendly and knowledgeable oral health assistant. Your purpose is to provide helpful, accurate information about oral health, dental hygiene, and common dental conditions. Keep responses concise, warm, and easy to understand. Always recommend seeing a dentist for diagnosis and treatment. Do not provide medical diagnoses.";

  const contents: ChatTurn[] = [
    { role: 'user', parts: [{ text: systemPrompt }] },
    { role: 'model', parts: [{ text: "Hello! I'm ORCare AI, your friendly oral health assistant. How can I help you today?" }] },
    ...history,
    { role: 'user', parts: [{ text: newMessage }] },
  ];

  const response = await fetch(`${GEMINI_URL}?key=${key}`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      contents,
      generationConfig: {
        temperature: 0.75,
        maxOutputTokens: 900,
      },
    }),
  });

  if (!response.ok) {
    const err = await response.text();
    throw new Error(`Gemini API error: ${err}`);
  }

  const data = await response.json();
  return data?.candidates?.[0]?.content?.parts?.[0]?.text ?? 'Sorry, I could not generate a response.';
}

File: app/src/main/java/com/simats/orcare/data/api/ApiService.kt
package com.simats.orcare.data.api

import com.simats.orcare.data.model.*
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface ApiService {
    @POST("api/auth/register")
    suspend fun register(@Body request: RegisterRequest): Response<RegisterResponse>

    @POST("api/auth/login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @POST("api/auth/verify-otp")
    suspend fun verifyOtp(@Body request: VerifyOtpRequest): Response<VerifyOtpResponse>

    @POST("api/auth/resend-otp")
    suspend fun resendOtp(@Body request: Map<String, String>): Response<Map<String, String>>

    @POST("api/auth/forgot-password")
    suspend fun forgotPassword(@Body request: ForgotPasswordRequest): Response<ForgotPasswordResponse>

    @POST("api/auth/reset-password")
    suspend fun resetPassword(@Body request: ResetPasswordRequest): Response<ResetPasswordResponse>

    @retrofit2.http.PUT("api/users/profile")
    suspend fun updateProfile(@Body request: UpdateProfileRequest): Response<UpdateProfileResponse>

    @retrofit2.http.DELETE("api/users/profile")
    suspend fun deleteProfile(): Response<Map<String, Any>>

    @POST("api/auth/request-delete-otp")
    suspend fun requestDeleteOtp(@Body request: RequestDeleteOtpRequest): Response<RequestDeleteOtpResponse>

    @POST("api/auth/confirm-delete-account")
    suspend fun confirmDeleteAccount(@Body request: ConfirmDeleteAccountRequest): Response<ConfirmDeleteAccountResponse>


    @retrofit2.http.GET("api/content/diseases")
    suspend fun getDiseases(): Response<List<DiseaseDto>>

    @POST("api/chat/save")
    suspend fun saveChatHistory(@Body request: ChatHistoryRequest): Response<ChatHistoryResponse>

    @retrofit2.http.GET("api/chat/history")
    suspend fun getChatHistory(): Response<List<RemoteChatSession>>

    @retrofit2.http.GET("api/content/learning")
    suspend fun getLearning(): Response<List<LearningCategoryDto>>

    @POST("api/content/feedback")
    suspend fun submitFeedback(@Body request: Map<String, String>): Response<Map<String, Any>>

    @POST("api/chat/ask")
    suspend fun askChatbot(@Body request: Map<String, String>): Response<Map<String, Any>>
}
